hcatalog/core/src/main/java/org/apache/hive/hcatalog/cli/SemanticAnalysis/CreateTableHook.java

```diff
@@ -64,16 +64,13 @@ public ASTNode preAnalyze(HiveSemanticAnalyzerHookContext context,
     // Analyze and create tbl properties object
     int numCh = ast.getChildCount();
 
-    tableName = BaseSemanticAnalyzer.getUnescapedName((ASTNode) ast
```

Comment:
The context for the changes in this file is that I am unsure this section of code ever worked and I am unsure of the intent.
Specifically - hcatalog tests fail because this starts throwing the exception I removed. The reason for this is because TOK_LIKETABLE was part of every single CREATE TABLE AST (you can see me remove the always TOK_LIKETABLE in my refactor of the grammar). I attempted to keep it the way it was to minimize diff and to keep logical changes together - but after a few hours of battling ANTLR I couldn't figure out a clean way to do it. (Suggestions are welcome)

Reply:
Having `TOK_LIKETABLE` in the AST even if it is not a `create table like` statement does not makes sense so I like your changes in the `CreateDDLParser.g`. 

I'm not familiar with hcatalog. When does `CreateTableHook` called? Could you please share a test?